priv/blogs/2022/07-29-observing-elixir-with-lightstep.md

```diff
@@ -0,0 +1,222 @@
+
+When I wake up in the morning, its time to figure out what went wrong, fix the bug, and get back to feature development. *Finding the root-cause of issues is hard*, especially when the issue is not a simple amendment to an [if/else/case expression](https://hexdocs.pm/elixir/1.13/Kernel.SpecialForms.html#case/2), but a performance regression in your infrastructure or database. In order to get to the bottom of these problems, we must have a way to understand our running system. We can achieve this by practicing observability and putting the right tools in place. 
+
+## What is observability?
+
+Observability is simply the ability to observe your running system, with the purpose of understanding its behavior. Forget the [three pillars](https://youtu.be/EJV_CgiqlOE), or anything else you might have read, observability is all about information, tools, and analysis to get to the bottom of your burning questions faster.
+
+If you've ever had questions like:
+
+* Why does my `/api/foo/thing` route suddenly have a P50 of triple its normal latency?
+* What is causing this memory spike in my application?
+* Is this production outage affecting all of my users?
+
+You might be in need of improving the observability of your system. Luckily, today we have standards, libraries, and tools available to answer the above questions much faster than before. 
+
+## Can I get observability with an APM like New Relic, AppSignal, or DataDog?
+
+Yes...and no. Application Performance Monitors (APMs) are a great way to get some observability of your system with little instrumentation. Out of the box, they will give you useful insights such as latency percentiles for each of the routes in your web application. From this metric data, you'll be able to see when you have a latency or rate spike on a particular route, due to increased traffic, a bug, or some infrastructure level problem. However, these tools only tell you **what** the problem is, they won't help you answer **why** it is happening, which is critical for the step of actually addressing any problems your application may be experiencing.
+
+## You want telemetry data at different levels of granularity
+
+In order to understand what's happening in your system, it's important to be able to see the big picture, and then zoom into various parts of your application to explore and discover regressions and issues. When I think about instrumenting my application with telemetry, I want the following kinds of data coming out of my system
+
+1. **Tracing** - [High-cardinality, high-dimensional](https://www.honeycomb.io/blog/so-you-want-to-build-an-observability-tool/) spans annotating the critical operations of my application (Helps me explore, discover, and correlate specific issues)
```

Comment:
I think it would be good to define cardinality beforehand

Reply:
Looks good! Thanks for the definition!

<img width="1475" alt="Screen Shot 2022-08-02 at 10 20 50 PM" src="https://user-images.githubusercontent.com/1019721/182510271-3ffdebb1-4b5d-49af-b3f9-8ca128d69713.png">